src/ui/public/vislib_vis_type/vislib_renderbot.js

```diff
@@ -46,7 +46,10 @@ module.exports = function VislibRenderbotFactory(Private) {
   VislibRenderbot.prototype.buildChartData = buildChartData;
   VislibRenderbot.prototype.render = function (esResponse) {
     this.chartData = this.buildChartData(esResponse);
-    this.vislibVis.render(this.chartData, this.uiState);
+    // to allow legend to render first
+    setTimeout(() => {
```

Comment:
i don't really like this (and its breaking tests). would returning a promise here be a good idea ?